### DISCUSSION TITLE
Webradios Export / Import functionality

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#74] [Feature Requests]
@rern

It would be outstanding to have complementary Webradio Export / Import Addons - trying to maintain Webradios on 4 systems is getting painful ;-)

Also would be great to have the ability to export / import the Webradio coverart images and Webradio bindings as well.

I am okay to manually setup the Webradio on the first system (Name, URL, and cover art), but I no longer have a way to transfer this setup to my other systems. In RuneAudio+R I used the backup / restore functionality to transfer the Webradio setup between systems, but this no longer works in rAudio 1.

I also tried to copy the webradios and webradioimg directories between two 7-zip sessions, but the URLs stored as filenames is changed during the copy process (thanks windows!). This is what eventually led me to the Webradio pls file import (to preserve the Webradio URL's in a pls file).

I am open to any suggestions / ideas that you might have.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

**Comment by R101G** [comment]
I keep a copy of the two folders from /srv/http/data for this. 

It would be useful to be able to select which settings apart from wifi are saved by the Backup/Restore feature.

**Reply by firstchanceongit** [reply]
Agree. It would also be good, for example, to be ablte to just export the library database (for synchronizing various rAudio devices in this regard).

**Reply by R101G** [reply]
You can install rsync to do this. There has to be some limit to the features an audio player has, although rern seems determined to disprove that :)

**Comment by ceejayemm** [comment]
Save your web radio information to individual .pls files (see attached example) and then use your favourite file transfer process to copy the stations you want to '/mnt/MPD/Webradio' on your rAudio installation (you might need to create this folder the first time around). Use the Web Radio import utility in the Addons page to import them.  Be aware of issue #178 at the moment which might cause you a problem if your urls contain an equals sign.

Chris
[BBC Radio 2.zip](https://github.com/rern/rAudio-1/files/5938686/BBC.Radio.2.zip)

**Comment by psklb** [comment]
Hi there,
I don't know if i'm at the right place to post a small script I build from importwebradio, just because I don't find any energy to produce one .pls file per radio. 

I found on this site a lot of radio streams (some of them are out of date, try before (or not) : http://wikradio.free.fr  
They are  french language radios  (like radiofrance package btw)
This site allow to export the selection you made in csv format.  One file for hundreds of radio if you want (and you can add yours)

I don't know how to make addons so I do it as a wild by scp my script and file directly in my rPi : 
(0. select radios from site and export csv file)
1. upload or scp script AND csv file in /mnt/MPD/Webradio/
2. log in rPI with ssh then cd in this directory then run script
3. appreciate by looking in library/webradios 

here is my script : 

```bash
#!/bin/bash

# all files include sub-directory
files=$( find /mnt/MPD/Webradio -maxdepth 1 -iname *.csv -type f 2> /dev/null )
if [[ -z $files ]]; then
	echo "No csv files found."
	echo 'put a csv from Wikradio.free.fr into /mnt/MPD/Webradio then run again.'
	exit
fi

# traitement du fichier csv importé de http://wikradio.free.fr 
echo "Csv Wikradio 2 Webradio Import ..."

readarray -t files <<<"$files"
for file in "${files[@]}"; do
	name=$( basename "$file" .csv )
    count=0
while IFS=";" read -r Pays nul1 Station Url Queue; do
  # do for each line in csv   
  [[ "$Pays" != "Pays" && "Station" != "" && "$Url" != "" ]] && echo $Station > /srv/http/data/webradios/${Url//\//|} && echo "$Station de $Pays Ajoutée" && ((++count))
done < $name.csv
  
echo "$count stations ajoutées ou mises à jour à partir du fichier $name"
done
count=$( ls -1q /srv/http/data/webradios | wc -l )
sed -i 's/\("webradio": \).*/\1'$count'/' /srv/http/data/mpd/counts
```

hope this can help, if only with your own csv file formatted like those created on wikradio (or at least with Station and Url at the same place in colomn order 

may be someone could convert this script in a really addon ?

Have a nice [what you want]

greetings
pascal
